compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -910,6 +944,60 @@ static FailureOr<SmallVector<Flow::DispatchWorkgroupsOp>> wrapInWorkgroupsOp(
   return result;
 }
 
+/// Create Flow::DispatchGroupsOps based on a fusion heuristic.
+static FailureOr<SmallVector<Flow::DispatchWorkgroupsOp>>
+createDispatchWorkgroups(AllOpsTrackingRewriter &rewriter,
+                         FunctionOpInterface funcOp,
+                         DominanceInfo const &dominanceInfo,
+                         bool generateWorkloadRegion) {
+  SmallVector<Flow::DispatchRegionOp> regionOps;
+
+  funcOp.walk([&](Flow::DispatchRegionOp op) { regionOps.push_back(op); });
+
+  // Clone additional producers and rewrite to DispatchWorkgroupsOp.
+  SmallVector<Flow::DispatchWorkgroupsOp> result;
+
+  // Find the root Op from the attribute and remove it
+  for (auto regionOp : regionOps) {
```

Comment:
I can maybe extend the dispatch.region so it can have an root op argument. I can fill it in the first pass. Here in the second pass, I can just call getRootOp. Does it make sense?

Reply:
Hopefully we wont need the root, but I see that you need it to get the workgroup count region. If we allow `flow.dispatch.region` to have a region operation, you could move the region creation to `flow.dispatch.region` creation right? Then the only part needed would be the `rewriteFlowDispatchRegionToFlowDispatchWorkgroups` ?
If so it might be better to just do that either as a separate PR that goes before this one or as part of this PR.